Dockerfile

```diff
@@ -1,18 +1,47 @@
-FROM python:3.8-slim-buster
+FROM python:3.12-slim as builder
+
+ENV LANG=C.UTF-8
+ENV PYTHONDONTWRITEBYTECODE=1
+ENV PYTHONUNBUFFERED=1
+ENV PATH="/app/venv/bin:$PATH"
+
+WORKDIR /app
+
 RUN apt-get update \
   && apt-get install -y --no-install-recommends procps gdb git curl \
-  && apt-get install -y gcc python3-dev \
+  && apt-get install -y gcc\
   && dpkg --add-architecture arm64 \
   && apt-get purge -y --auto-remove \
   && rm -rf /var/lib/apt/lists/*
+RUN pip install poetry==1.6.1
 
-RUN curl -sSL https://install.python-poetry.org | python3 -
-RUN /root/.local/bin/poetry config virtualenvs.create false
 COPY poetry.lock pyproject.toml /app/
-WORKDIR /app/
-RUN /root/.local/bin/poetry install --no-interaction --no-root
 
 COPY src /app/src
-RUN /root/.local/bin/poetry install
 
-CMD exec /bin/bash -c "trap : TERM INT; sleep infinity & wait"
\ No newline at end of file
+RUN python -m venv /app/venv && \
+    . /app/venv/bin/activate && \
+    poetry config virtualenvs.create false && \
+    poetry install --no-dev
+
+FROM python:3.12-slim
+
+RUN apt-get update \
+  && apt-get install -y --no-install-recommends procps gdb git curl \
```

Comment:
Why do we need git?
It has a critical CVE

Reply:
todo, see what else we dont need